tech/post/2017-09-25-Kubernetes-Scheduler.md

```diff
@@ -0,0 +1,145 @@
+# Kubernetes Scheduler
+
+## 1. Scheduler Introducation
+
+The Kubernetes Scheduler runs as a process alongside the other master components such as the API server. Its interface to the API server is to watch for Pods with an empty ```PodSpec.NodeName```, and for each Pod, it posts a binding indicating where the Pod should be scheduled.
+
+![](../images/2017-09-25-Kubernetes-Scheduler/Scheduler.png)
+
+## 2. Scheduler Workflow
+
+The Kubernetes Scheduler is focused on scheduling pods into specified work nodes by schedule algorithm, which also be called ```Bind```.The input of Scheduler is the pods which need to be scheduled and the nodes which could be scheduled into, and the output of Scheduler is ```Bind``` which connects between pods and nodes.
+
+![](../images/2017-09-25-Kubernetes-Scheduler/Workflow.png)
+
+The Kubernetes Scheduler tries to find a node for each Pod, one at a time.
+- First it applies a set of "predicates" to filter out inappropriate nodes. For example, if the PodSpec specifies resource requests, then the Scheduler will filter out nodes that don't have at least that much resources available.

```

Comment:
nodes that don't have at least that much resources available/ nodes that don't have enough available resources?

Reply:
done